# Sample size calculation

Calculation of sample size based on the type of the metrics that you need to analyse.

In [1]:
from scipy import stats

## Example of Significance Power Calculator

In [2]:
def binomial_sample_size(metric, mde, alpha, beta):
    # standard normal distribution to determine z-values
    snd = stats.norm(0, 1)

    Z_beta = snd.ppf(1-beta)
    print(Z_beta)

    Z_alpha = snd.ppf(1-alpha/2)
    print(Z_alpha)

    # average of probabilities from both groups
    p = (metric + metric+mde) / 2
    print(p)

    N = (2 * p * 
             (1 - p) * 
             (0.84 + 1.96)**2
             / mde**2)

    return N

In [3]:
binomial_sample_size(metric=0.33, mde=0.02, alpha=0.05, beta=0.2)

0.8416212335729143
1.959963984540054
0.34


8796.479999999998

In [5]:
def continous_sample_size(metric, mde, sd, alpha, beta):
    # standard normal distribution to determine z-values
    snd = stats.norm(0, 1)

    Z_beta = snd.ppf(1-beta)
    print(Z_beta)

    Z_alpha = snd.ppf(1-alpha/2)
    print(Z_alpha)

    N = (2 * sd**2 * 
             (Z_beta + Z_alpha)**2
             / mde**2)

    return N

In [7]:
continous_sample_size(metric=30673, mde=100, sd=91, alpha=0.05, beta=0.2)

0.8416212335729143
1.959963984540054


12.99931461602896

**Note**: In case we want to determine `mde=100` users difference for our Daily Active User `metric=30673`, we'll need `12.99931461602896` (13) days.

In [8]:
continuos_sample_size(metric=30673, mde=300, sd=91, alpha=0.05, beta=0.2)

0.8416212335729143
1.959963984540054


1.4443682906698845

**Note**: If we want to determine something closer to `100%` relative difference then we'll need `1.4443682906698845` (2) days.